### ISSUE TITLE
Static candidates that found by Sonar somehow missed by our Check

### ISSUE THREAD

**Issue by romani** [sevntu-checkstyle/sevntu.checkstyle#403]
http://nemo.sonarqube.org/component_issues?id=com.puppycrawl.tools%3Acheckstyle#resolved=false|severities=MINOR|createdAfter=2015-10-28T07%3A03%3A11%2B0100

@Vladlis , please recheck why StaticCandidateCheck does not catch this.

**Comment by Vladlis** [comment]
@romani, in both cases constants from TokenTypes are used and we can not cannot claim that the calls are static as we have no information about TokenTypes.
Here's an example of a false-positive which could have appeared:
```java

class ConstantsClass {
    public static final int METHOD_DEF = 1;
}

class Issue403 {
    public ConstantsClass TokenTypes = new ConstantsClass();
}

class Child extends Issue403 {
    private void foo() {
        int a = TokenTypes.METHOD_DEF;
    }
}
```
Of course it's a rare case, but still we cannot rely on the naming convention.

**Comment by romani** [comment]
Yes, you are right.
